components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/AbstractHTTPFunction.java

```diff
@@ -0,0 +1,174 @@
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.conn.ConnectTimeoutException;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.wso2.carbon.identity.application.authentication.framework.AsyncProcess;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.JsGraphBuilder;
+import org.wso2.carbon.identity.conditional.auth.functions.common.utils.ConfigProvider;
+import org.wso2.carbon.identity.conditional.auth.functions.common.utils.Constants;
+
+import java.io.IOException;
+import java.net.SocketTimeoutException;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Abstract class for handling http calls.
+ */
+public abstract class AbstractHTTPFunction {
+
+    private static final Log LOG = LogFactory.getLog(AbstractHTTPFunction.class);
+    protected static final String TYPE_APPLICATION_JSON = "application/json";
+    private static final char DOMAIN_SEPARATOR = '.';
+    private final List<String> allowedDomains;
+
+    private CloseableHttpClient client;
+
+    public AbstractHTTPFunction() {
+
+        RequestConfig config = RequestConfig.custom()
+                .setConnectTimeout(ConfigProvider.getInstance().getConnectionTimeout())
+                .setConnectionRequestTimeout(ConfigProvider.getInstance().getConnectionRequestTimeout())
+                .setSocketTimeout(ConfigProvider.getInstance().getReadTimeout())
+                .build();
+        client = HttpClientBuilder.create().setDefaultRequestConfig(config).build();
+        allowedDomains = ConfigProvider.getInstance().getAllowedDomainsForHttpFunctions();
+    }
+
+    protected void executeHttpMethod(HttpUriRequest request, Map<String, Object> eventHandlers) {
+
+        AsyncProcess asyncProcess = new AsyncProcess((context, asyncReturn) -> {
+            JSONObject json = null;
+            int responseCode;
+            String outcome;
+            String epUrl = null;
+
+            if (request.getURI() != null) {
+                epUrl = request.getURI().toString();
+            }
+
+            if (!isValidRequestDomain(request.getURI())) {
+                outcome = Constants.OUTCOME_FAIL;
+                LOG.error("Provided Url does not contain a allowed domain. Invalid Url: " + epUrl);
+                asyncReturn.accept(context, Collections.emptyMap(), outcome);
+                return;
+            }
+
+            try (CloseableHttpResponse response = client.execute(request)) {
+                responseCode = response.getStatusLine().getStatusCode();
+                if (responseCode == 200) {
```

Comment:
Can we consider 201 also as a SUCCESS response code?

Reply:
what about 202?

Reply:
fixed. considered the whole 200 range